packages/suite/src/utils/suite/parseUri.ts

```diff
@@ -1,75 +1,78 @@
-import { PROTOCOL_SCHEME } from '@suite-support/Protocol';
+import { PROTOCOL_SCHEME } from '@suite-reducers/protocolReducer';
+import { isNetworkSymbol } from '@wallet-utils/accountUtils';
+import type { Network } from '@wallet-types';
 
-/* eslint-disable prefer-destructuring */
-export interface ParsedURI {
-    address: string;
-    amount?: string;
-}
-
-const stripPrefix = (str: string): string => {
-    if (!str.match(':')) {
-        return str;
+// Parse URL query string (like 'foo=bar&baz=1337) into an object
+export const parseQuery = (uri: string) => {
+    const params: Record<string, string | undefined> = {};
+    try {
+        const index = uri.indexOf('?');
+        new URLSearchParams(uri.substring(index)).forEach((v, k) => {
+            params[k] = v;
+        });
+    } catch (e) {
+        // empty
     }
-    const parts = str.split(':');
-    parts.shift();
-    return parts.join('');
+    return params;
 };
 
-// Parse URL query string (like 'foo=bar&baz=1337) into an object
-export const parseQuery = (str: string) =>
-    str
-        .split('&')
-        .map(val => val.split('='))
-        .reduce((vals: { [key: string]: any }, pair: string[]) => {
-            if (pair.length > 1) {
-                vals[pair[0]] = pair[1];
-            }
-            return vals;
-        }, {});
-
 // Parse a string read from a bitcoin QR code into an object
-export const parseUri = (uri: string): ParsedURI => {
-    const str = stripPrefix(uri);
-    const query: string[] = str.split('?');
-    const values: Record<string, any> = query.length > 1 ? parseQuery(query[1]) : {};
-    const address = query[0] || '';
-
-    return {
-        ...values,
-        address,
-    };
+export const parseUri = (uri: string) => {
+    try {
+        return new URL(uri);
+    } catch (e) {
+        // silent
+    }
 };
 
-interface BaseProtocol {
-    scheme: string;
+export type BitcoinProtocolInfo = {
```

Comment:
I would not call it `BitcoinProtocolInfo` but just `Coin...` Litecoin, Dogecoin,... can be easily added in few minutes.

Reply:
https://github.com/trezor/trezor-suite/pull/4572/commits/a56666794df19596b50a1c5759b2b14b8a750c12